2.6/root/usr/share/container-scripts/mongodb/replset-supervisor.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+#
+# This script registers the current container into MongoDB replica set or initiate replSet if $1 == "initiate"
+
+source ${CONTAINER_SCRIPTS_PATH}/base-functions.sh
+source ${CONTAINER_SCRIPTS_PATH}/init-functions.sh
+source ${CONTAINER_SCRIPTS_PATH}/replset-functions.sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+set -x
+
+wait_mongo "UP"
+
+if [[ "$1" == "initiate" ]]; then
+  # Prepare master
+  echo "=> Initiating the replSet ${MONGODB_REPLICA_NAME} ..."
+
+  mongo_initiate
+
+  echo "=> Creating MongoDB users ..."
+  mongo_create_admin
+  mongo_create_user "-u admin -p ${MONGODB_ADMIN_PASSWORD}"
+
+  # Add one pod to this replSet to be able to address it with mongodb service
+  member_addr=$(echo "$(endpoints)" | head -n 1)
+  echo "=> Adding first member ${member_addr} to replSet ..."
+  mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --eval "rs.add('${member_addr}:27017');"
+
+  mongo_wait_replset "-u admin -p ${MONGODB_ADMIN_PASSWORD}" "$(mongo_addr)"
+
+  echo "=> Successfully initialized replSet ..."
+
+  # Wait for at least 3 members
+  members=0
+  while [ "${members}" -le 3 ] 2>/dev/null || [[ $? -ne 1 ]] ; do
```

Comment:
If `members` contains non-numeric value test will print an error message therefore there is `2>/dev/null`. And in that case we still wants to wait so the `[[ $? -ne 1 ]]` is there to indicate that the previous test failed with $?=2 (contains non-numeric value)...